docs/puzzles/pooling.md

```diff
@@ -0,0 +1,320 @@
+---
+id: pooling
+title: Pooling
+---
+
+The way that Chia Network does pooling is unlike any blockchain that has come before it. Pool operators actually rely on an on-chain smart contract to verify that they will be able to directly claim any potential pool rewards that farmers create.  This allows pools to trust farmers enough to pay them out, while still keeping the power of making the blocks in the hands of the farmer.  This means that the decentralization of the network remains the same, even as the rewards get concentrated to the pool!
+
+In this section, we're going to break down how all of this works in Chialisp.  This section assumes you have already read the section about [singletons](https://chialisp.com/docs/puzzles/singletons) (or at least understand how they work) as that is the outer contract that wraps the pooling puzzles.
+
+## Design Requirements
+
+There are a few requirements that were set for how the pooling protocol would work on the Chia Network. Let's go over them now:
+
+* **The farmer still farms the blocks.** This is incredibly important for network decentralization. If this is not true, then the bigger a pool gets, the closer it is to gaining 51% of the network resources.  We still want the farmers to create the blocks, but we need some way to assure the pool that the farmer will give them the reward when they do.  We do this by creating plots that farm directly to a specific puzzle hash, and ensuring that puzzle hash is something that pool can claim.
+* **The farmer must still be able to change pools.** Initially, this seems to conflict with the first requirement.  Plots can be made to send rewards directly to a puzzle hash, but you cannot change that puzzle hash once they are plotted.  If you want to switch pools, you'll have to remake all of your plots! We solve this by having our plots create payments that a specific singleton (also called a "plot nft") can claim.  Then, we can lend partial control of that singleton to a pool, but still retain the ability to reclaim our singleton and lend it instead to a different pool.  Our plots will remain effective as long as we retain control of the singleton.
```

Comment:
The term "singleton" isn't explained here. Maybe this term is established earlier, but if it isn't, it might be worth just describing it as a "smart contract"

Reply:
The top of the page says that it assumes knowledge of singletons if that counts.